src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -228,11 +228,77 @@ public ISqlConnectionFactory ConnectionFactory
         public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connectionInfo) => this.OwnerToConnectionMap.TryGetValue(ownerUri, out connectionInfo);
 
         /// <summary>
-        /// Validates the given ConnectParams object. 
+        /// Refreshes the auth token of a given connection 
         /// </summary>
-        /// <param name="connectionParams">The params to validate</param>
-        /// <returns>A ConnectionCompleteParams object upon validation error, 
-        /// null upon validation success</returns>
+        /// <param name="ownerUri">The URI of the connection</param>
+        /// <returns>true upon successful refresh of the auth token,
+        /// false if the uri cannot be found, refresh is not needed, or
+        /// if refresh fails </returns>
+
+        internal async Task<bool> TryRefreshAuthToken(string ownerUri)
+        {
+            ConnectionInfo connInfo;
+            if (this.TryFindConnection(ownerUri, out connInfo))
+            {
+                // If not an azure connection, no need to refresh token
+                if (connInfo.ConnectionDetails.AuthenticationType != "AzureMFA") 
+                {
+                    return false;
+                } 
+                else 
+                {
+                    // Check if token is expired
+                    var maxTolerance = 2 * 60; // two minutes
+                    if (connInfo.ConnectionDetails.ExpiresOn - DateTimeOffset.Now.ToUnixTimeSeconds() < maxTolerance)
+                    {   
+                        var requestMessage = new RefreshTokenParams
+                        {
+                            AccountId = connInfo.ConnectionDetails.GetOptionValue("azureAccount", string.Empty),

```

Comment:
If we don't have an accountId or "authority" (tenant) then we shouldn't even try to request a refresh - since that's guaranteed to fail.